hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerAttribute.java

```diff
@@ -189,7 +189,11 @@ public boolean remove(T key, ContainerID value) {
     Preconditions.checkNotNull(key);
 
     if (this.attributeMap.containsKey(key)) {
-      return Collections.unmodifiableNavigableSet(this.attributeMap.get(key));
+      NavigableSet<ContainerID> returneSet = new TreeSet<>();
+      TreeSet<ContainerID> tmpSet = (TreeSet<ContainerID>)
```

Comment:
Is the cast necessary?

```suggestion
      NavigableSet<ContainerID> tmpSet =
```